rules-java-project/tests/src/test/java/org/jboss/windup/project/operation/test/ProjectWithHintTest.java

```diff
@@ -172,7 +193,25 @@ public void perform(GraphRewrite event, EvaluationContext context, FileLocationM
                     .begin()
                     .addRule()
                     .when(Project.dependsOnArtifact(Artifact.withArtifactId("abc")))
-                    .perform(Hint.titled("Test Hint").withText("Test Hint Body").withEffort(42).and(addMatch));
+                    .perform(Hint.titled("Test Hint").withText("Test Hint Body").withEffort(42).and(addMatch))
+                    .addRule()
+                    .when(Project.dependsOnArtifact(Artifact.withArtifactId("{param}")))
+                    .perform(Hint.titled("Test {param}").withText("Test {param} Body").withEffort(42).and(addMatch))
+                    .where("param").matches("abc")
+                    .addRule()
+                    .when(Project.dependsOnArtifact(Artifact.withGroupId("{param}")))
+                    .perform(Hint.titled("Test {param}").withText("Test {param} Body").withEffort(42).and(addMatch))
+                    .where("param").matches("xyz")
+                    // deliberately not matching rule to have a evaluationStrategy.modelSubmissionRejected() case
+                    .addRule()
+                    .when(Project.dependsOnArtifact(Artifact.withGroupId("{param}").andArtifactId("{param}")))
```

Comment:
i don't think I understand this. Is it a check for an artifact with the same groupId and artifactId? Or does it substitute in order from a list of parameters?

Reply:
Yep, it searches for a dependency with the same value for groupId and artifactId and, as you can imagine, it won't be found so this tests the `modelSubmissionRejected()` method.